File: Java/1409_Queries_on_a_Permutation_With_Key.java
class Solution {
    public int[] processQueries(int[] queries, int m) {
        List<Integer> P = new ArrayList<>();
        int[] result = new int[queries.length];
        for(int i = 0; i < m; i++){
            P.add(i+1);
        }
        for(int i = 0; i < queries.length; i++){
            int target = P.indexOf(queries[i]);
            result[i] = target;
            P.remove(target);
            P.add(0, queries[i]);
            /*
            for(int j = 0; j < m; j++){
                if(queries[i] == P.get(j)){
                    result[i] = j;
                    P.remove(j);
                    P.add(0, queries[i]);
                    //System.out.println(P);
                }
            }
            */
        }
        return result;
    }
}

File: CS/1409_Queries_on_a_Permutation_With_Key.cs
public class Solution {
    public int[] ProcessQueries(int[] queries, int m) {
        List<int> P = new List<int>();
        int[] result = new int[queries.Length];
        for(int i = 1; i <= m; i++){
            P.Add(i);
        }
        for(int i = 0; i < queries.Length; i++){
            result[i] = P.IndexOf(queries[i]);
            P.RemoveAt(result[i]);
            //P.Remove(queries[i]);
            P.Insert(0, queries[i]);
        }
        return result;
    }
}
